crypto/rand/rand_lib.c

```diff
@@ -211,22 +224,40 @@ int RAND_set_rand_engine(ENGINE *engine)
     CRYPTO_THREAD_unlock(rand_engine_lock);
     return 1;
 }
-# endif
+#  endif
+# endif /* OPENSSL_NO_DEPRECATED_3_0 */
 
 void RAND_seed(const void *buf, int num)
 {
+    EVP_RAND_CTX *drbg;
+# ifndef OPENSSL_NO_DEPRECATED_3_0
     const RAND_METHOD *meth = RAND_get_rand_method();
 
-    if (meth != NULL && meth->seed != NULL)
+    if (meth != NULL && meth->seed != NULL) {
         meth->seed(buf, num);
+        return;
+    }
+# endif
+
+    drbg = RAND_get0_primary(NULL);
+    if (drbg != NULL && num > 0)
+        EVP_RAND_reseed(drbg, 0, NULL, 0, buf, num);
 }
 
 void RAND_add(const void *buf, int num, double randomness)
 {
+    EVP_RAND_CTX *drbg;
+# ifndef OPENSSL_NO_DEPRECATED_3_0
     const RAND_METHOD *meth = RAND_get_rand_method();
 
-    if (meth != NULL && meth->add != NULL)
+    if (meth != NULL && meth->add != NULL) {
         meth->add(buf, num, randomness);
+        return;
+    }
+# endif
+    drbg = RAND_get0_primary(NULL);
+    if (drbg != NULL && num > 0)
+        EVP_RAND_reseed(drbg, 0, NULL, 0, buf, num);
```

Comment:
So `randomess` is ignored? Is that the correct behaviour?

Reply:
It's been like this for quite a while.  We trust out sources which we always use, we don't trust what the user supplies.  I can see arguments both ways.

In a FIPS context, we are required to ignore this.